rules/RFC/Project_Cypress_Development_Rules.md

```diff
@@ -0,0 +1,106 @@
+  fixtures/             # typed JSON fixtures that store test data
+  support/
+    cy-commands/        # *.ts files with custom commands that cannot be inside a scenario or a page object, e.g. a DB manipulation command
+    glue-endpoints/     # *.ts files each of which contains ulrs, payload, test data references and headers for one specific API endpoint that can be reused inside a test
+    page-objects/       # *.ts files each of which contains a page object - selectors and actions that are limited to a single application page
+    scenarios/          # *.ts files that contain complex reusable scenarios that involve actions with more than one page object, so two or more application pages
+    e2e.ts              # global setup, test libs
+    index.d.ts          # TS augmentation for Cypress
+```
+
+## TypeScript & Config
+
+- For TS config use existing `tsconfig.json`
+- For Cypress config use existing `cypress.config.ts`
+- for E2E config use existing `cypress/support/e2e.ts`
+
+## Selectors & Page Objects
+
+- MUST use one of these attributes for element selection when available: id, name, data-qa
+- All selectors that belong to one application page and methods that use them MUST be created inside the relevant page object
+- Each application page MUST exist as a separate page object in cypress/support/page-objects
+- Stub external/unstable requests with cy.intercept; avoid real third-party calls in CI.
+- If the new feature adds new elements on a page that already exists, e.g. shopping list feature adds `add to shopping list` button on PDP (Product Details Page), then you should update existing page object for PDP. The project should have exactly 1 page object for Shopping list details page and exactly 1 page object for PDP. There should be no additional page objects that deal with elements on either of these pages.
+
+## Test Specs
+Test specs should not have hardcoded urls, test data or element selectors. Page objects should provide tech spec with selectors and urls, fixtures should provide the test data.
+
+Test specs MUST NOT contain any direct cy.get(), cy.find(), or other Cypress element selection commands. All element interactions MUST be performed through page object getters and methods. Test specs should only contain:
+- Page object method calls (e.g., pageObject.getButton().click())
+- Assertions (e.g., pageObject.getElement().should('be.visible'))
+- Test flow logic (e.g., cy.visit(), cy.url().should())
+
+Forbidden in test specs:
+- cy.get('[data-qa="..."]')
+- cy.find('[data-qa="..."]')
+- Any direct element selection
+
+Required approach:
+- Create getters in page objects for all elements
+- Use page object methods for all interactions
+- Test specs should read like business logic, not technical implementation
+
+## Flake Prevention
+- Hardcoded wait times MUST NOT be used in tests to improve reliability and reduce flakiness - DO NOT use `cy.wait()` where it can be avoided.
```

Comment:
Maybe we can reuse something from 'core' rules, or it doesn't make sense? 
https://github.com/spryker-dev/prompt-library/blob/main/rules/cypress/cypress-tests.md?plain=1#L113